### PR TITLE
tests for target exponent

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [ ] made this PR as small as possible
- [ ] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- Bug Fixes
  - Tightened exponent-adjustment overflow guards to address edge-case scaling behavior.

- Tests
  - Added comprehensive unit tests covering target-exponent adjustments: large diffs, max overflow, scaling up/down, and expected revert scenarios.

- Performance
  - Updated gas benchmark snapshots with minor measurement changes across decimal-float tests.

### CODE REVIEW

File: src/lib/implementation/LibDecimalFloatImplementation.sol [hunk 1]
```diff
@@ -623,124 +619,128 @@ library LibDecimalFloatImplementation {
...
+            // all powers of 10 look like 1 with a different exponent
+            if (signedCoefficient == 1e76) {
+                return (exponent + 76, 0);
+            }
+            bool isAtLeastE76 = signedCoefficient >= 1e76;
+
+            // This is a positive log. i.e. log(x) where x >= 1.
+            if (exponent >= (isAtLeastE76 ? -76 : -75)) {
+                int256 y1Coefficient;
+                int256 y2Coefficient;
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply src/lib/implementation/LibDecimalFloatImplementation.sol]: it isn't in an unchecked block
